numpy/core/src/multiarray/descriptor.c

```diff
@@ -451,18 +451,18 @@ _convert_from_array_descr(PyObject *obj, int align)
         }
         else if (PyTuple_Check(name)) {
             if (PyTuple_GET_SIZE(name) != 2) {
-                _report_generic_error();
+                PyErr_Format(PyExc_TypeError, "PyObject within tuple needs to be a 2-tuple, has [%d] elements", PyTuple_GET_SIZE(name));
```

Comment:
Perhaps _"If a tuple, the first element of a field tuple must have two elements"_, or some similar message that indicates that non-tuples can also be ok.

Reply:
Thanks for the suggestion, I will incorporate in my next commit.